### PR TITLE
Epoch resharding

### PR DESCRIPTION
**Pull request by LeoHChen**

### CODE REVIEW

File: core/blockchain.go [hunk 1]
```diff
@@ -74,7 +74,10 @@ const (
...
 	//  state mutation bug and 2) implement key passphrase recycle across
 	//  process restart (exec) for shard migration
-	BlocksPerEpoch = 1000000000000
+	// TODO lc - set to 10800, aproximiately 1 day per epoch based on 8
+	// seconds per block
+	// BlocksPerEpoch = 10800
+	BlocksPerEpoch = 5
 
 	// BlockChainVersion ensures that an incompatible database forces a resync from scratch.
 	BlockChainVersion = 3
```
Comment (chaosma) [inline core/blockchain.go]: we have 2 versions already?

File: consensus/consensus_service.go [hunk 2]
```diff
@@ -166,6 +166,24 @@ func (consensus *Consensus) DebugPrintPublicKeys() {
...
+		consensus.CommitteePublicKeys[pubKey.SerializeToHexStr()] = true
+	}
+	prepareBitmap, err := bls_cosi.NewMask(consensus.PublicKeys, consensus.LeaderPubKey)
+	if err == nil {
+		consensus.prepareBitmap = prepareBitmap
+	}
+	consensus.pubKeyLock.Unlock()
+	return len(consensus.PublicKeys)
+}
+
```
Comment (chaosma) [inline consensus/consensus_service.go]: We can use UpdateBlsPublicKeys to replace similar code in UpdatePublicKeys

File: core/resharding.go [hunk 3]
```diff
@@ -256,5 +271,43 @@ func GetInitShardState() types.ShardState {
 		}
 		shardState = append(shardState, com)
 	}
+	utils.GetLogInstance().Info("GetInitShardState", "shardState", shardState)
+	return shardState
+}
+
+// GetSwapOneShardState returns the initial shard state at genesis.
+func GetSwapOneShardState() types.ShardState {
```
Comment (chaosma) [inline core/resharding.go]: As per our discussion,  we will have a single function, which read shard state at end of each epoch. And it should read from db using key=epoch.

**Comment by LeoHChen** [comment]
no longer needed.

**Pull request closed without merging**